ELEMENTS-1327: unlock polymer version

richardsd opened the pull request:
In short, for the data table the selection problem comes from two code paths where **this.$.list.toggleSelectionForIndex** is called:
- [onCheckboxTap handler](https://github.com/nuxeo/nuxeo-elements/blob/4df9e4d26c473da55b9fbc4f5f3c80b4375e8cad/ui/nuxeo-data-table/iron-data-table.js#L852)
- [_selectionHandler](https://github.com/nuxeo/nuxeo-elements/blob/maintenance-3.0.x/ui/nuxeo-data-table/iron-data-table.js#L573) that is registered as listener in the [page-provider-display-behavior](https://github.com/nuxeo/nuxeo-elements/blob/4df9e4d26c473da55b9fbc4f5f3c80b4375e8cad/ui/nuxeo-page-provider-display-behavior.js#L413) (when `selectOnTap` is set to false)

richardsd committed:
ELEMENTS-1327: unlock polymer version

nuxeojenkins commented:
View issue in JIRA: [ELEMENTS-1327](https://jira.nuxeo.com/browse/ELEMENTS-1327): Unlock @polymer/polymer dependency on Storybook

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/nuxeo/nuxeo-elements?pullRequest=475) <br/>All committers have signed the CLA.

richardsd committed:
ELEMENTS-1327: keep selection behavior when selectOnTap is set

Prevent multiple selections of same item in data table's internal iron-list.

Fix _isSelected method when multiSelection is disabled.

mnixo commented on ui/nuxeo-data-table/iron-data-table.js:
```diff
@@ -849,7 +849,7 @@ import '../nuxeo-button-styles.js';
     _onCheckBoxTap(e) {
       if (this.selectionEnabled && !this.selectAllActive) {
-        this.$.list.toggleSelectionForIndex(e.model.index);
+        // this.$.list.toggleSelectionForIndex(e.model.index);
```
Should we add more information about why we're commenting this line out?

richardsd replied:
I think we should remove the line and maybe add a comment explaining the behavior. wdyt?

richardsd committed:
ELEMENTS-1327: add selectOnTap option to data table stories

richardsd committed:
ELEMENTS-1327: set multiSelection by default in data table stories

richardsd committed:
ELEMENTS-1447: disable select all in data table if multiSelection is inactive

richardsd commented:
Cross repo build https://github.com/nuxeo/nuxeo-web-ui/runs/4534055450 🔵

Gabez0r commented on ui/nuxeo-page-provider-display-behavior.js:
```diff
@@ -389,11 +389,21 @@ export const PageProviderDisplayBehavior = [
     _isSelected(item) {
-      return !!(this.selectedItems && this.selectedItems.length && this.selectedItems.indexOf(item) > -1);
+      return this.multiSelection
+        ? !!(this.selectedItems && this.selectedItems.length && this.selectedItems.indexOf(item) > -1)
+        : !!(this.selectedItem && this.selectedItem === item);
     },
 
     _isIndexSelected(index) {
-      return !!(this.selectedItems && this.selectedItems.length && this.selectedItems.indexOf(this.items[index]) > -1);
+      return this.multiSelection
+        ? !!(
+            this.selectedItems &&
+            this.selectedItems.length &&
+            this.items &&
+            this.items.length > index &&
+            this.selectedItems.indexOf(this.items[index]) > -1
+          )
+        : !!(this.selectedItem && this.items && this.items.length > index && this.selectedItem === this.items[index]);
```
Should we add some unit tests for this?

richardsd replied:
Maybe it's a good idea 👍  Let me see how we can improve it.

Gabez0r reviewed (commented):
LGTM.

richardsd committed:
ELEMENTS-1327: add unit tests for single selection and helpers in data table

richardsd commented:
Added unit tests to cover the selection helpers update.
